app/screens/send-bitcoin-screen/improvement-request.tsx

```diff
@@ -14,14 +15,51 @@ export const SuggestAnImprovement: React.FC = () => {
     theme: { colors },
   } = useTheme()
   const [isActive, setIsActive] = React.useState(false)
+  const [token, setToken] = React.useState("")
   const [improvement, setImprovement] = React.useState("")
 
   const dismiss = React.useCallback(() => {
     setIsActive(false)
   }, [setIsActive])
 
-  const submitImprovement = () => {
-    // call to mattermost API and backend function
+  const submitImprovement = async () => {
+    const savedToken = await AsyncStorage.getItem("mattermostToken")
+    if (savedToken === "") {
+      const response = await fetch("https://chat.galoy.io/api/v4/users/login", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          // eslint-disable-next-line camelcase
+          login_id: "username",
+          password: "password",
+        }),
+      })
+
+      if (response.ok) {
+        const newToken = response.headers.get("Token")
+        setToken(newToken ?? "")
+        await AsyncStorage.setItem("mattermostToken", newToken ?? "")
+      }
+    } else {
+      setToken(savedToken ?? "")
+    }
+
+    await fetch("https://chat.galoy.io/api/v4/posts", {
+      method: "POST",
```

Comment:
this would have to be from the backend. we can't embed securely a token or username/password in a mobile app

Reply:
oh okay, then should I make a PR in the galoy repo?

Reply:
I would not advise for it.. there is a steep learning curve on the backend so maybe console.log() the answer for now and someone with knowledge of the backend can then make the appropriate mutation.

Reply:
What would be the best way to know whether it's the first payment or not?

Reply:
you can fetch the list of transactions